os/mm/mm_heap/mm_sem.c

```diff
@@ -125,7 +130,14 @@ void mm_seminitialize(FAR struct mm_heap_s *heap)
 
 int mm_trysemaphore(FAR struct mm_heap_s *heap)
 {
-	pid_t my_pid = getpid();
+	pid_t my_pid;
+
+	if (heap == NULL) {
```

Comment:
This function is called from umm_trysemaphore, kmm_trysemaphore.
The umm function verifies heap first and then call this function so that it does not have any problem.
But the kmm doesn't and it may causes null dereference.
We should keep the same policy for checking null.
If you want to check null here, you should remove heap checking from umm. It's duplicated.

Reply:
Please check others in this point of view.